ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/sandbox/services/reset/ResetServiceDatabaseIT.scala

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.services.reset
+
+import java.sql.{Connection, DriverManager}
+
+import anorm.SqlParser._
+import anorm.{SQL, SqlStringInterpolation}
+import com.digitalasset.platform.sandbox.services.reset.ResetServiceDatabaseIT.countRowsOfAllTables
+import com.digitalasset.platform.sandbox.services.{DbInfo, SandboxFixture}
+import com.digitalasset.platform.store.DbType
+import com.digitalasset.resources.ResourceOwner
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+abstract class ResetServiceDatabaseIT extends ResetServiceITBase with SandboxFixture {
+
+  "ResetService" when {
+
+    "run against a database backend" should {
+
+      "leave the tables in the expected state" in {
+
+        val ignored = Set(
+          "flyway_schema_history", // this is not touched by resets, it's used for migrations
+          "packages" // preserved by the reset to match the compiled packages still loaded in the engine
+        )
+
+        for {
+          ledgerId <- fetchLedgerId()
+          _ <- reset(ledgerId)
+          counts <- countRowsOfAllTables(ignored, database.get)
+        } yield {
+
+          val expectedToHaveOneItem = Set(
+            "parameters", // a new set of parameters is stored at startup
+            "participant_command_completions", // using static time, one checkpoint is going to be saved
+            "ledger_entries" // using static time, one checkpoint is going to be saved
+          )
```

Comment:
This can be on the same level as `ignored` to avoid computing it every iteration.

Reply:
I would prefer to keep it closer to the tests.

Reply:
Then why not have three test cases?

Reply:
So that exceptions to the "all tables are empty" rule are grouped together in one place. I don't think that re-initializing a set multiple times is an issue in a test where we shoot 10 commands, wait for completions and reset the ledger. :wink: But thanks for the feedback.

Reply:
So that exceptions to the "all tables are empty" rule are grouped together in one place. I don't think that re-initializing a set multiple times is an issue in a test where we shoot 10 commands, wait for completions and reset the ledger. :wink: But thanks for the feedback.